plugin/src/Reconciler/fetchInstances.lua

```diff
@@ -0,0 +1,73 @@
+local Selection = game:GetService("Selection")
+
+local Rojo = script:FindFirstAncestor("Rojo")
+local invariant = require(script.Parent.Parent.invariant)
+
+local Log = require(Rojo.Packages.Log)
+
+local function fetchInstances(idList, instanceMap, apiContext)
+	return apiContext:fetch(idList):andThen(function(body: { sessionId: string, path: string })
+		-- The endpoint `api/fetech/idlist` returns a table that contains
+		-- `sessionId` and `path`. The value of `path` is the name of a
+		-- file in the Content folder that may be loaded via `GetObjects`.
+		local objects = game:GetObjects("rbxasset://" .. body.path)
+		-- `ReferentMap` is a folder that contains nothing but
+		-- ObjectValues which will be named after entries in `instanceMap`
+		-- and have their `Value` property point towards a new Instance
+		-- that it can be swapped out with. In turn, `reified` is a
+		-- container for all of the objects pointed to by those instances.
```

Comment:
Not sure ReferentMap is necessary - it seems a little complex for what it's actually accomplishing

Because our serializers provide ordering guarantees for children, and GetObject insertion reflects this ordering, I think the server could construct a model file with children in the same order as the requested IDs, and then the client can match them back up via this ordering. 

Alternatively, if it's uncomfortable to rely on child order this way, then the server could write multiple files, and respond with multiple paths (again, in the same order as the requested IDs), at the expense of multiple file system writes and multiple calls to GetObjects

Reply:
I went with ReferentMap because it was the first solution I thought of that avoided a bunch of FS writes and reads, not necessarily because I thought it was the best option. In practice it's not *really* all that complicated, it just sounds bad when you write it out. I can get behind wanting to simplify it as much as possible though, since it's unusual for Rojo.

There's the potential to write thousands of files to a file system and then load them in very quickly if we use multiple paths. I'm not inherently opposed to that, but I'd want to check what the performance of that was. It's not great as-is with one monolith model, I can imagine it getting really bad with a bunch of them.

I'm going to veto relying on child ordering just on principle. It makes me feel gross.

Reply:
I think I overlooked the additional complication caused by instances with parent restrictions, so ordering wouldn't work anyway... it's probably fine